internal/parse/arguments.go

```diff
@@ -0,0 +1,28 @@
+package parse
+
+import (
+	"flag"
+)
+
+type Options struct {
+	ReverseFlag bool
+	Files       []string
+}
+
+func optionFlags(ops *Options) {
+	flag.BoolVar(&(ops.ReverseFlag), "reverse", false, "reverse the result of comparisons")
+	flag.BoolVar(&(ops.ReverseFlag), "r", false, "reverse the result of comparisons")
+	flag.Parse()
+}
+
+func ArgumentParsing() *Options {
+	ops := Options{}
+	optionFlags(&ops)
+	ops.Files = flag.Args()
+	if len(ops.Files) == 0 {
```

Comment:
Note: in the Unix world `-` means `STDIN` as well. For instance `sort -r -` means: "sort lines from STDIN in a reverse order"

Reply:
That is fixed, didn't know about that